test/test_bad_schema_ref.rb

```diff
@@ -19,7 +19,7 @@ def test_bad_uri_ref
     }
 
     data = [1,2,3]
-    assert_raises(Errno::ENOENT) do
+    assert_raises(JSON::Schema::ReadFailed) do
```

Comment:
I think it would be good to test that `location` and `type` are set correctly in both error cases. @iainbeeston do you have an idea how this could be easily integrated in our test helper setup?